service/storage_service.cc

```diff
@@ -5747,10 +5747,6 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
             throw std::runtime_error("Migrating within the same node is not supported");
         }
 
-        if (src == dst) {
-            co_return;
-        }
-
```

Comment:
This change will conflict with intra-node migration support, which removes the check for host above.
I think the right fix is to move `src == dst` check above the host check.